tests/neg/i12432.scala

```diff
@@ -0,0 +1,4 @@
+package dotty.tools.dotc.typer
+
+class i1 ( i1 : annotation ) // error
```

Comment:
The bug report in #12432 is for a `case class`:

```scala
case class i1 ( i1 : annotation )
```

which still crashes with this PR.